backend/src/index.ts

```diff
@@ -1,28 +1,66 @@
+/*
+ * Main entry point for the backend server.
+ * 
+ * On import of `app`, an instance of the Express application is created. This 
+ * file manages the server startup and shutdown logic.
+ */
+
 /* eslint-disable sort-imports, import/first */
 
-// Load environment variables
 import dotenv from 'dotenv';
 dotenv.config();
 
 import app from './app.js';
 import fs from 'fs';
 import https from 'https';
+import { Server } from 'http';
 
 const port = process.env.PORT;
+export let server: Server<any, any>;
 
 if (process.env.NODE_ENV !== 'production') {
-  app.listen(port, () => {
+  // Start the server in development or test mode
+  server = app.listen(port, () => {
     console.log(`\nExpress listening on port ${ port }`);
   });
+  
 } else {
-  const privateKey = fs.readFileSync(process.env.PRIVATE_KEY_PATH || '', 'utf8');
-  const certificate = fs.readFileSync(process.env.CERTIFICATE_PATH || '', 'utf8');
-  const ca = fs.readFileSync(process.env.CA_PATH || '', 'utf8');
+  // Start the server in production mode
+  const { PRIVATE_KEY_PATH, CERTIFICATE_PATH, CA_PATH } = process.env;
+
+  if (!PRIVATE_KEY_PATH || !CERTIFICATE_PATH || !CA_PATH) {
```

Comment:
I appreciate the explicit configuration checks and clear error message! Very helpful